plugins/reporters/fossid/src/main/kotlin/FossIdSnippetReporter.kt

```diff
@@ -38,6 +38,9 @@ class FossIdSnippetReporter : Reporter by delegateReporter {
     override val type = "FossIdSnippet"
 
     override fun generateReport(input: ReporterInput, outputDir: File, options: Map<String, String>): List<File> {
+        val hasFossIdResults = input.ortResult.scanner?.scanResults?.any { it.scanner.name == "FossId" } ?: false
+        check(hasFossIdResults) { "No FossID scan results have been found." }
```

Comment:
This now makes me think again about the general question how a reporter should be have in case of empty / invalid input. Failing like this IMO makes the reporter command exit with a code > 0, which will probably break your CI pipeline. And that might be a problem if you simply always activate FossID reports, even for pipeline runs that do not use FossID as a scanner.

Reply:
> Should I change something ?

Not necessarily, I just wanted to put this up for consideration. Maybe you could cross-check how other reporters behave in this regard. I have a hunch that they do *not* fail, but print some custom text into the report instead.

But actually, if you keep this code, `check` should become `require` (as you're verifying input, not state).